File: 0118.杨辉三角/0118-杨辉三角.cpp
class Solution {
public:
    vector<vector<int>> generate(int numRows) {;
        vector<vector<int>> res;
        vector<int> temp;
        if (numRows <= 0)
            return res;
        temp.push_back(1);
        res.push_back(temp);
        temp.clear();
        for (int i = 1;i < numRows; ++i){
            temp.clear();
            temp.push_back(1);
            for (int j = 1; j < i; ++j){
                temp.push_back(res[i-1][j-1] + res[i-1][j]);
            }
            temp.push_back(1);
            res.push_back(temp);
        }        
        return res;
    }
};

File: 0118.杨辉三角/0118-杨辉三角.py
class Solution(object):
    def generate(self, numRows):
        """
        :type numRows: int
        :rtype: List[List[int]]
        """
        if not numRows:
            return []
        res = [[1]]
        for i in range(1, numRows):
            tmp = [1]
            for j in range(1, i):
                tmp.append(res[i-1][j - 1] + res[i - 1][j])
            tmp.append(1)
            res.append(tmp[:])
        return res